src/blueprint/server.rs

```diff
@@ -160,6 +164,22 @@ fn validate_hostname(hostname: String) -> Valid<IpAddr, String> {
     }
 }
 
+fn validate_apollo(apollo: Option<Apollo>) -> Valid<Option<Apollo>, String> {
+    match apollo {
+        Some(apollo) => {
+            let is_valid = regex::Regex::new(r"[A-Za-z0-9-_]@[A-Za-z0-9-_]")
+                .unwrap()
+                .is_match(&apollo.graph_ref);
+            if is_valid {
+                Valid::succeed(Some(apollo))
+            } else {
+                Valid::fail("Invalid value in graph_ref".to_string())
```

Comment:
Need unit tests. 
Update error message, such that it gives hints on how to fix it.